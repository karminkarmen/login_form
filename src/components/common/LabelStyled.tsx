import styled from 'styled-components';

type Props = {
    target: string;
    isUppercased: boolean;
}

const LabelStyled = styled.label.attrs((props: Props) => ({
    htmlFor: props.target
}))<Props>`
    margin: ${props => props.theme.margins.normal};
    color: ${props => props.theme.colors.main};
    text-transform: ${props => props.isUppercased ? 'uppercase' : 'none'};
    font-weight: bold;
    letter-spacing: ${props => props.theme.letterSpacing.normal};
    font-size: ${props => props.theme.fontSizes.normal};
    font-family: ${props => props.theme.fontFamilies.main}, sans-serif;
`;

export default LabelStyled;
