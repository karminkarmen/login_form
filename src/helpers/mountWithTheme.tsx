import React from 'react';
import { mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import theme from '../configs/theme';

const mountWithTheme = (child: React.ReactElement) => {
    return mount(child, {
        wrappingComponent: ({ children }) => <ThemeProvider theme={theme}>{children}</ThemeProvider>,
    });
}

export default mountWithTheme;