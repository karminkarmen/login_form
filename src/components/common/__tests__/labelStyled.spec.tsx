import React from 'react';
import mountWithTheme from '../../../helpers/mountWithTheme';
import LabelStyled from '../LabelStyled';

describe('<LabelStyled />', () => {

    const props: React.ComponentProps<typeof LabelStyled> = {
        target: 'remember',
        isUppercased: true,
    };

    it('matches snapshot', () => {
        const wrapper = mountWithTheme(<LabelStyled {...props} />);
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('should render label html tag with specified for and target properties', () => {
        const wrapper = mountWithTheme(<LabelStyled {...props} />);

        expect(wrapper.find(`label[target='${props.target}']`)).toHaveLength(1);
        expect(wrapper.find(`label[target='${props.target}']`)).toHaveLength(1);
    });

});
