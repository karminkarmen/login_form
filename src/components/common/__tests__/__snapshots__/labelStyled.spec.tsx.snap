// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LabelStyled /> matches snapshot 1`] = `
"<styled.label target=\\"remember\\" isUppercased={true}>
  <label target=\\"remember\\" htmlFor=\\"remember\\" className=\\"sc-AxjAm eNUZro\\" />
</styled.label>"
`;
